src/main/scala/mesosphere/marathon/core/deployment/impl/TaskReplaceActor.scala

```diff
@@ -164,16 +165,16 @@ class TaskReplaceActor(
         val goal = if (runSpec.isResident) Goal.Stopped else Goal.Decommissioned
         instanceTracker.setGoal(instance.instanceId, goal, GoalChangeReason.Upgrading)
           .pipeTo(self)
-      } // 2) An old and decommissioned instance was successfully killed
-      else if (considerTerminal(condition) && instance.state.goal.isTerminal()) {
+      } // 2) An old and decommissioned instance was successfully killed (or was never launched in the first place if condition == Scheduled)
```

Comment:
Third change: 
- if the _old_ instance was decommissioned (`goal.isTerminal == true`) but 
- it was only _Scheduled_ `considerTerminal(condition) == false` 
then this if-condition will not match. Hence the somewhat ugly `|| condition == Condition.Scheduled` addition.

Reply:
so this is basically:
- old instance we did not plan to replace yet but since it's terminal, we're going to decommission it (that's 1)
- old instance that will be expunged (pretty much duplicating what InstanceDeleted would give us) (that's part of 2)
- old unreachable instance that was decommissioned (that's the other part of 2)

I spent some time trying to figure out how to express this better since this extra if is not very fancy :( But I did not come up with anything super fancy so far. Let me think a bit more :)